### PR TITLE
Clean up signature to MessageDispatch/dispatch,

### PR DESCRIPTION
**Pull request by stevelr**
Change return type of MessageDispatch/dispatch to Vec<u8> and remove (now) unnecessary lifetime qualifiers.
Also some uses of dispatch use &Message as the third parameter and others use Message. This change makes them consistently Message.

Bumps wasmbus-rpc to 0.10.0 and weld-codegen to 0.5.0.

This is a breaking api change, but the dispatch function is almost exclusively used within generated code.

Signed-off-by: stevelr <steve@cosmonic.com>

### CODE REVIEW

File: macros/src/lib.rs
Comment (thomastaylor312) [inline macros/src/lib.rs:198]: Nit: While we are here, all of these types should have fully qualified paths. So this should be `::wasmbus-rpc::Context` and `Message` should be something similar

File: rpc-rs/Cargo.toml
Comment (thomastaylor312) [inline rpc-rs/Cargo.toml:3]: Were there any other changes that we needed to bring in before we bump?

File: rpc-rs/src/provider.rs
Comment (thomastaylor312) [inline rpc-rs/src/provider.rs:2]: Nit: I generally like adding the allows exactly where they are needed unless we are talking about something more than 3 of these similar lints

**Review by thomastaylor312 — APPROVED** [review]

File: macros/src/lib.rs
Comment (stevelr) [reply macros/src/lib.rs:198]: done

File: rpc-rs/Cargo.toml
Comment (stevelr) [reply rpc-rs/Cargo.toml:3]: We don't have to publish immediately if there are other things we're waiting on. This change is not super-urgent

Comment (stevelr) [reply rpc-rs/Cargo.toml:3]: I wouldn't be surprised if there is an update to async-nats in a couple weeks that we'll want to push ..

File: rpc-rs/src/provider.rs
Comment (stevelr) [reply rpc-rs/src/provider.rs:2]: me too ... but I removed this line and don't get clippy warnings any more so it's out!

File: rpc-rs/Cargo.toml
Comment (brooksmtownsend) [reply rpc-rs/Cargo.toml:3]: We should take a look and see if we can get any async-nats fixes before we publish. It is true that once we merge this PR we'd have to cherry-pick if we needed to release a patch version to `0.9`

**Pull request merged**
